distribution/docs/src/main/jdocs/content/_developingComponents/developing-plugins-contents.adoc

```diff
@@ -0,0 +1,403 @@
+
+. Import the ${branding} interface packages to the bundle manifest (in addition to any other required packages). +
+`Import-Package: ddf.catalog,ddf.catalog.plugin,ddf.catalog.operation,ddf.catalog.event`
+
+. Export the service to the OSGi registry. +
+*Blueprint descriptor example* +
+`<service ref="[[SamplePreDeliveryPlugin]]"interface="ddf.catalog.plugin.PreDeliveryPlugin" />`
+
+====== Pre-Delivery Plugin Failure Behavior
+
+In the event that this Catalog Plugin cannot operate but does not wish to fail the transaction, a `PluginExecutionException` will be thrown.
+For any other Exceptions, the Catalog will "fail safe" and the Operation will be cancelled.
+If processing is to be explicitly stopped, a `StopProcessingException` will be thrown.
+
+====== Invoking Pre-Delivery Plugins
+
+Pre-delivery plugins are invoked serially, prioritized by descending OSGi service ranking.
+The plugin with the highest service ranking will be executed first. 
+
+The output of a pre-delivery plugin is sent to the next pre-delivery plugin, until all have executed and the Delivery Method is invoked on the associated Subscription.
+
+===== Implementing Pre-Subscription Plugins
+
+Develop a custom Pre-Subscription Plugin.
+
+. Create a Java class that implements `PreSubscriptionPlugin`. +
+`public class SamplePreSubscriptionPlugin *implements* ddf.catalog.plugin.PreSubscriptionPlugin
+
+. Implement the required method.
+ * `public Subscription process(Subscription input) *throws* PluginExecutionException, StopProcessingException;`
+
+====== Pre-Subscription Plugin Failure Behavior
+
+In the event that this Catalog Plugin cannot operate but does not wish to fail the transaction, a `PluginExecutionException` will be thrown.
+For any other Exceptions, the Catalog will "fail safe" and the Operation will be cancelled.
+If processing is to be explicitly stopped, a `StopProcessingException` will be thrown.
+
+====== Invoking Pre-Subscription Plugin
+
+Pre-subscription plugins are invoked serially, prioritized by descending OSGi service ranking. 
+That is, the plugin with the highest service ranking will be executed first. 
+
+The output of a pre-subscription plugin is sent to the next pre-subscription plugin, until all have executed and the create Subscription operation is sent to the Event Processor.
+
+===== Implementing Pre-Resource Plugins
+
+Develop a custom Pre-Resource Plugin.
+
+. Create a Java class that implements `PreResourcePlugin`.
+`public class SamplePreResourcePlugin *implements* ddf.catalog.plugin.PreResourcePlugin`
+
+. Implement the required method. +
+ * `public ResourceRequest process(ResourceRequest input) *throws* PluginExecutionException, StopProcessingException;`
+
+. Import the ${branding} interface packages to the bundle manifest (in addition to any other required packages). +
+`Import-Package: ddf.catalog,ddf.catalog.plugin,ddf.catalog.operation`
+
+. Export the service to the OSGi registry.
+.Blueprint descriptor example
```

Comment:
It's been a while since I formatted this originally, but I think it was because some are part of numbered lists, which can be finicky in terms of accidentally restarting the numbers, but I'll experiment with adding the `[source...]` syntax where possible because the code decoration is nicer to look at.